pkg/build/builder/sti.go

```diff
@@ -92,8 +92,14 @@ func (s *S2IBuilder) Build() error {
 	if s.build.Spec.Strategy.SourceStrategy == nil {
 		return fmt.Errorf("the source to image builder must be used with the source strategy")
 	}
-
+	var runtimeImageName string
 	var push bool
+	runtimeImage := s.build.Spec.Strategy.SourceStrategy.RuntimeImage
+	runtimeArtifacts := copyArtifactSourceList(s.build.Spec.Strategy.SourceStrategy.RuntimeArtifacts)
+
+	if runtimeImage != nil && len(runtimeImage.Name) > 0 {
+		runtimeImageName = runtimeImage.Name
+	}
```

Comment:
Maybe we need to add validation on the OpenShift side. I have validation on s2i side but I'm not sure how it will work with OpenShift and also it's better to fail fast.